crates/router/src/connector/paybox.rs

```diff
@@ -292,10 +302,8 @@ impl ConnectorIntegration<api::PSync, types::PaymentsSyncData, types::PaymentsRe
         event_builder: Option<&mut ConnectorEvent>,
         res: Response,
     ) -> CustomResult<types::PaymentsSyncRouterData, errors::ConnectorError> {
-        let response: paybox::PayboxPaymentsResponse = res
-            .response
-            .parse_struct("paybox PaymentsSyncResponse")
-            .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
+        let response: paybox::PayboxSyncResponse =
+            paybox::parse_url_encoded_to_syn_struct(res.response)?;
```

Comment:
Remove redundant functions, the conversion to json_string can be handled by one function and custom operations on top of it can be written in the respective flows